docs/architecture/devops-for-aspnet-developers/actions-deploy.md

```diff
@@ -0,0 +1,491 @@
+
+1. You can also now see deployments. Navigate to `https://{your repository url}/deployments` to view your deployments:
+
+    - ![View deployments](./media/actions/deploy/deployments.jpg)
+    **Figure 8**: View deployments.
+
+## Deploy to PROD
+
+Now that you've deployed successfully to `PRE-PROD`, you'll want to deploy to `PROD`. Deployment to `PROD` will be slightly different since you don't need to copy the website again - you just need to swap the `staging` slot with the production slot. you'll do this using an Azure cli (`az`) command.
+
+1. Navigate to the `.github/workflows/dotnet.yml` file and select the pencil icon to edit the file.
+1. Add a new job below the `deploy_staging` job as follows:
+
+    ```yml
+          run: az logout  # <-- last line of previous job: insert below this line
+
+      deploy_prod:
+        needs: deploy_staging
+        runs-on: ubuntu-latest
+      
+        environment:
+          name: PROD
+          url: ${{ steps.slot_swap.outputs.url }}
+        
+        steps:
+        - name: Login via Azure CLI
+          uses: azure/login@v1
+          with:
+            creds: ${{ secrets.AZURE_CREDENTIALS }}
+        
+        - name: Swap staging slot into production
+          id: slot_swap
+          run: |
+            az webapp deployment slot swap -g ${{ env.rg-name }} -n ${{ env.app-name }} -s staging
+            url=$(az webapp show -g ${{ env.rg-name }} -n ${{ env.app-name }} --query "defaultHostName" -o tsv)
+            echo "::set-output name=url::http://$url"
+          
+        - name: az cli logout
+          run: az logout
+    ```
+
+    Notice the following things:
+  
+    1. Once again, you specify a new job `deploy_prod` that `needs` `deploy_staging` to complete before starting.
+    1. You're targeting the `PROD` environment this time. Also, the `url` value is different from before.
+    1. For the steps, you don't need to download the artifact since you're just going to perform a slot swap. You start by executing a `login` to the Azure context.
+    1. The `Swap staging slot into production` step is a multi-line `run` command (note the use of the pipe symbol `|`). You also specify an `id` for this step so that you can refer to it (you refer to it in the `url` property of the `environment`). The first line executes the slot swap using the variables you defined above in the workflow. The second line uses an `az webapp show` command to extract the URL of the target web app. This final line uses `::set-output` in an echo to create an output variable for this task, setting the value to the web app URL.
+
+    > [!NOTE]
+    > The URL _must_ start with `http://` or `https://` or it will not render.
+
+1. Commit the file.
+1. Let the workflow run for a couple minutes until it has deployed to `PRE-PROD`. At this point the workflow will pause and wait for the required approval since you're targeting the `PROD` environment, which requires an approval as defined earlier:
```

Comment:
```suggestion
1. Let the workflow run for a couple minutes until it has deployed to `PRE-PROD`. At this point, the workflow will pause and wait for the required approval since you're targeting the `PROD` environment, which requires an approval as defined earlier:
```